### PR TITLE
fivetran-destination: Set application name for SQL client

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR sets the application name of the client the Fivetran destination uses to send requests to Materialize to `mz_fivetran_destination`. It also adds the name to the set of application names we tag Prometheus metrics with.

### Motivation

More easily debug and track usage of the Fivetran destination

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

File: src/sql/src/session/hint.rs [hunk 1]
```diff
@@ -54,6 +58,8 @@ impl ApplicationNameHint {
             "dbt" => ApplicationNameHint::Dbt(Private),
             "web_console" => ApplicationNameHint::WebConsole(Private),
             "mz_psql" => ApplicationNameHint::MzPsql(Private),
+            // Note: Make sure this is kept in sync with the `fivetran-destination` crate.
+            "mz_fivetran_destination" => ApplicationNameHint::MzFivetranDestination(Private),
```
Comment (benesch) [inline src/sql/src/session/hint.rs:62]: I'd go with the more verbose `materialize_fivetran_destination` to be explicit; the `mz` in `mz_psql` is to indicate the `mz` CLI and not "Materialize" broadly.

```suggestion
            "materialize_fivetran_destination" => ApplicationNameHint::MzFivetranDestination(Private),
```

File: src/sql/src/session/hint.rs [hunk 2]
```diff
@@ -45,6 +47,8 @@ pub enum ApplicationNameHint {
     WebConsole(Private),
     /// Request came from the `psql` shell spawned by `mz`.
     MzPsql(Private),
+    /// Request came from our Fivetran Destination,
+    MzFivetranDestination(Private),
```
Comment (benesch) [inline src/sql/src/session/hint.rs:51]:
```suggestion
    MaterializeFivetranDestination(Private),
```

File: src/fivetran-destination/src/destination/config.rs [hunk 3]
```diff
@@ -19,6 +19,8 @@ use crate::fivetran_sdk::{
     ConfigurationFormResponse, ConfigurationTest, FormField, TestRequest, TestResponse, TextField,
 };
 
+pub const FIVETRAN_DESTINATION_APPLICATION_NAME: &str = "mz_fivetran_destination";
```
Comment (benesch) [inline src/fivetran-destination/src/destination/config.rs:22]:
```suggestion
pub const FIVETRAN_DESTINATION_APPLICATION_NAME: &str = "materialize_fivetran_destination";
```

**Review by benesch — APPROVED** [review]

**Review by benesch — APPROVED** [review]

File: src/sql/src/session/hint.rs (hunk 1 above)
Comment (ParkMyCar) [reply src/sql/src/session/hint.rs]: Makes sense, updated!

**Pull request merged**